atlasdb-client/src/main/java/com/palantir/atlasdb/schema/SchemaDependentTableMetadata.java

```diff
@@ -46,18 +49,56 @@
     public SchemaMetadataPersistence.SchemaDependentTableMetadata persistToProto() {
         SchemaMetadataPersistence.SchemaDependentTableMetadata.Builder builder =
                 SchemaMetadataPersistence.SchemaDependentTableMetadata.newBuilder();
-        builder.setCleanupRequirement(cleanupRequirement());
+        cleanupMetadata().accept(new CleanupMetadataPersisterVisitor(builder));
         return builder.build();
     }
 
     public static SchemaDependentTableMetadata hydrateFromProto(
             SchemaMetadataPersistence.SchemaDependentTableMetadata message) {
-        CleanupRequirement cleanupRequirement = CleanupRequirement.ARBITRARY_ASYNC; // strictest default
-        if (message.hasCleanupRequirement()) {
-            cleanupRequirement = message.getCleanupRequirement();
+        ImmutableSchemaDependentTableMetadata.Builder builder = ImmutableSchemaDependentTableMetadata.builder();
+        builder.cleanupMetadata(parseCleanupMetadata(message));
+        return builder.build();
+    }
+
+    private static CleanupMetadata parseCleanupMetadata(
+            SchemaMetadataPersistence.SchemaDependentTableMetadata message) {
+        switch (message.getCleanupMetadataCase()) {
+            case ARBITRARYCLEANUPMETADATA:
```

Comment:
itshardtoparsethingswithoutspaces.

this reads like a [Banjo-Kazooie sandcastle code](http://banjokazooie.wikia.com/wiki/Cheat_Codes) - let's use `SCREAMING_SNAKE_CASE` for readability.

Reply:
These names aren't unfortunately as poetic though.
(I've seen the sandcastle grid before, but not what the cheats actually are!)

- This table is lean; no need to clean. (Null)
- This is a Stream Store; I'll need to clean more. (Stream-Store)
- I don't know what to do - maybe you do. (Arbitrary)

The cases here are actually generated by the protobuf compiler. This would involve making the proto field names snake case as [that's how the code gen works](https://developers.google.com/protocol-buffers/docs/reference/java-generated#oneof). 

Maybe it's worth the trade-off of having a few lines of snake case in the protobuf? It looks like we're bound to have a bit of inconsistency somewhere, though in the protobuf it's guaranteed to only occur once.